src/OpenLoco/Graphics/PaletteMap.h

```diff
@@ -1,58 +1,29 @@
 #pragma once
 
+#include "../Core/Span.hpp"
 #include "Colour.h"
 #include "ImageId.h"
 #include <array>
 #include <cstdint>
 #include <optional>
 
-namespace OpenLoco::Gfx
+namespace OpenLoco::Gfx::PaletteMap
 {
+    /**
+     * The typical palette map size used to render images.
+     */
+    constexpr size_t kDefaultSize = 256;
 
     /**
      * Represents an 8-bit indexed map that maps from one palette index to another.
      */
-    struct PaletteMap
-    {
-    private:
-        uint8_t* _data{};
-        uint32_t _dataLength{};
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wunused-private-field"
-        uint16_t _numMaps;
-#pragma clang diagnostic pop
-        uint16_t _mapLength;
-
-    public:
-        static const PaletteMap& getDefault();
-
-        PaletteMap() = default;
-
-        PaletteMap(uint8_t* data, uint16_t numMaps, uint16_t mapLength)
-            : _data(data)
-            , _dataLength(numMaps * mapLength)
-            , _numMaps(numMaps)
-            , _mapLength(mapLength)
-        {
-        }
+    template<std::size_t TSize>
+    using Buffer = std::array<PaletteIndex_t, TSize>;
 
-        template<std::size_t TSize>
-        PaletteMap(uint8_t (&map)[TSize])
-            : _data(map)
-            , _dataLength(static_cast<uint32_t>(std::size(map)))
-            , _numMaps(1)
-            , _mapLength(static_cast<uint16_t>(std::size(map)))
-        {
-        }
+    using View = stdx::span<const PaletteIndex_t>;
 
-        uint8_t& operator[](size_t index);
-        uint8_t operator[](size_t index) const;
-        uint8_t* data() const { return _data; }
-        uint8_t blend(uint8_t src, uint8_t dst) const;
-        void copy(size_t dstIndex, const PaletteMap& src, size_t srcIndex, size_t length);
-    };
+    View getDefault();
 
-    std::optional<uint32_t> getPaletteG1Index(ExtColour paletteId);
-    std::optional<PaletteMap> getPaletteMapForColour(ExtColour paletteId);
-    std::optional<PaletteMap> getPaletteMapFromImage(const ImageId image);
+    std::optional<View> getForColour(ExtColour paletteId);
+    std::optional<View> getFromImage(const ImageId image);
```

Comment:
What about renaming these to just `get` and using function overloading for the two versions